ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -465,6 +490,328 @@ const triggerSet: TriggerSet<Data> = {
+          en: 'Drop light tower',
+        },
+        darkTower: {
+          en: 'Drop dark tower',
+        },
+      },
+    },
+    {
+      id: 'P12S Engravement 1 Tower Soak',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravementTiltIds },
+      condition: (data, matches) => data.engravementCounter === 1 && data.me === matches.target,
+      suppressSeconds: 5, // avoid second (incorrect) alert when debuff switches from soaking tower
+      alertText: (data, matches, output) => {
+        if (data.engravement1Towers.indexOf(data.me) === -1) {
+          // Did not drop a tower, so needs to soak one.
+          if (matches.effectId === engravementIdMap.lightTilt)
+            return output.lightTilt!();
+          return output.darkTilt!();
+        }
+      },
+      outputStrings: {
+        lightTilt: {
+          en: 'Soak dark tower',
+        },
+        darkTilt: {
+          en: 'Soak light tower',
+        },
+      },
+    },
+    // In Engravement 2 (Superchain 1), all supports or  DPS will receive lightTilt and darkTilt (2 each).
+    // All 4 also receive Heavensflame Soul.
+    // The other role group will receive lightTower, darkTower, lightBeam, and darkBeam.
+    // To resolve the Beams during the 2nd orb, lightBeam needs to stack with darkTower and both darkTilts, and vice versa.
+    // After the 3rd orb, lightTower and darkTower will drop their towers, and  darkBeam and lightBeam (respectively) will soak them.
+    // The four Heavensflame players all simultaneously need to spread to drop their AoEs.
+    // Debuffs do change based on mechanic resolution, which can complicate things:
+    // - When a lightTilt player soaks a dark beam, their debuff will change to darkTilt, and vice versa.
+    // - Once the beams detonate, the lightBeam debuff disappears and is replaced with lightTilt (same with dark).
+    // So only use the initial debuff to resolve the mechanic, and use a long suppress to avoid incorrect later alerts.
+    {
+      id: 'P12S Engravement 2 Beam Soak',
+      type: 'GainsEffect',
+      netRegex: { effectId: Object.values(engravementIdMap) },
+      condition: (data, matches) => data.engravementCounter === 2 && data.me === matches.target,
+      delaySeconds: 5, // wait until first orb resolves so as not to confuse
+      durationSeconds: 7, // keep active until beams go off
+      suppressSeconds: 30,
+      response: (_data, matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          lightBeam: {
+            en: 'Light Beam (stack with Dark group)',
+          },
+          darkBeam: {
+            en: 'Dark Beam (stack with Light group)',
+          },
+          lightTower: {
+            en: 'Light Tower (stack with light group)',
+          },
+          darkTower: {
+            en: 'Dark Tower (stack with dark group)',
+          },
+          lightTilt: {
+            en: 'Stack (light group) - spread later',
```

Comment:
```suggestion
            en: 'Light Tilt => spread later',
```

I think this is less wordy.

Reply:
Responded above before I saw this.  That works, although I think I would say "Light Stack => spread later".  Tilt is the technical name of the debuff (well, actually Astral Tilt and Umbral Tilt), but I'm honestly not sure how many people know that it's called Tilt, outside from us nerds.

Reply:
Yeah, that's fair; mostly I think there's no "stack" person there, which was a little of my sticking point.  `Light Tilt` or `Light Debuff` (which is maybe vague) sounded better, since the most important part is the "Light" word.

Reply:
> Yeah, that's fair; mostly I think there's no "stack" person there, which was a little of my sticking point. `Light Tilt` or `Light Debuff` (which is maybe vague) sounded better, since the most important part is the "Light" word. (Edited to add: `Light Stack` also fine if you don't think it's confusing too and maybe it's just me. Just trying to disambiguate as much as possible and hard to know what other people think!)

Hmm, that's fair given how we use 'Stack' elsewhere.  How about 'Light Group => spread later'?

Reply:
Sounds good to me!